app/models/text.rb

```diff
@@ -1,5 +1,14 @@
 class Text < ApplicationRecord
+  has_many :reads, dependent: :destroy
+  has_many :read_users, through: :reads, source: :user
   validates :genre, presence: true
   validates :title, presence: true
   validates :content, presence: true
+  
+  def read_by?(user)
+    reads.find_by(user_id: user.id).present?
+  end
+  def next
+    Text.where("id > ?", self.id).order("id ASC").first
```

Comment:
これですと存在しないときにエラーが出ると思います。

```suggestion
    Text.where("id > ?", self.id).order("id ASC")&.first
```